tests/logging/mpi_log_all_ranks.cpp

```diff
@@ -12,35 +12,33 @@ bool trace_msg_in_file(Logger<FP>& logger) {
   return (logger.should_log(Log::level::trace)) and (logger.get_file_sink()->should_log(Log::level::trace));
```

Comment:
So, this only checks that a log and sink should both log, but does not trace if a msg is found in the file, right?

You already spent a lot of time on this, so I'm ok merging. Later, we could do stuff like this:

```
TEST_CASE(blah) {
 ...
  std::string trace_msg = "trace msg";
  std::string debug_msg = "debug msg";
  std::string log_file;
  {
     Logger<...> mylog(...);
     mylog.set_level(Log::level::debug);
     log_file = mylog.logfile_name();
     mylog.trace(trace_msg); // Should not show up
     mylog.debug(debug_msg); // Should show up
  }
  std::ifstream ifile(log_file);
  std::stringstream ss;
  ss << ifile.rdBuf();
  REQUIRE(ss.str().find(trace_msg)==std::string::npos); // Trace output NOT present
  REQUIRE(ss.str().find(debug_msg)!=std::string::npos); // Debug output present  

}
```

Reply:
I'd be happy to put that in.  I'm off tomorrow, so it would be next week.